### PR TITLE
Upgrading to Node.js 18 and AWS SDK v3

### PR DESCRIPTION
**Pull request by andreaswittig**

### CODE REVIEW

File: operations/alb-access-logs-anonymizer.yaml [hunk 1]
```diff
@@ -172,55 +172,51 @@ Resources:
...
+            } catch(err) {
+              if (err.Code === 'NoSuchKey') {
+                console.log('original object no longer exists, checking for anonymized object')
+                try {
+                  await s3.send(new HeadObjectCommand({
+                    Bucket: record.s3.bucket.name,
+                    Key: anonymizedKey
+                  }));
+                } catch(err) {
+                  if (err.Code === 'NoSuchKey') {
```
Comment (michaelwittig) [inline operations/alb-access-logs-anonymizer.yaml:206]: what about all the other error Codes? Before, an error was always thrown. Now just NoSuchKey errors are thrown?

File: operations/cloudfront-access-logs-anonymizer.yaml [hunk 2]
```diff
@@ -173,55 +173,51 @@ Resources:
...
+            } catch(err) {
+              if (err.Code === 'NoSuchKey') {
+                console.log('original object no longer exists, checking for anonymized object')
+                try {
+                  await s3.send(new HeadObjectCommand({
+                    Bucket: record.s3.bucket.name,
+                    Key: anonymizedKey
+                  }));
+                } catch(err) {
+                  if (err.Code === 'NoSuchKey') {
```
Comment (michaelwittig) [inline operations/cloudfront-access-logs-anonymizer.yaml:207]: what about all the other error Codes? Before, an error was always thrown. Now just NoSuchKey errors are thrown?

File: state/s3.yaml [hunk 3]
```diff
@@ -425,9 +425,9 @@ Resources:
               }
             }
             if (event.RequestType === 'Delete') {
-              s3.getBucketAcl({
+              s3.send(new GetBucketAclCommand({
                 Bucket: event.ResourceProperties.Bucket
-              }, (err, acl) => {
+              }), (err, acl) => {
                 if (err) {
                   if (err.code === 'NoSuchBucket') {
```
Comment (michaelwittig) [inline state/s3.yaml:432]: sometimes you use err.Code, sometimes stay with err.code?

File: operations/cloudfront-access-logs-anonymizer.yaml (hunk 2 above)
Comment (andreaswittig) [reply operations/cloudfront-access-logs-anonymizer.yaml]: Fixed!

File: state/s3.yaml (hunk 3 above)
Comment (andreaswittig) [reply state/s3.yaml]: It's `err.name`. :D

**Pull request merged**